### PR TITLE
Move FiniteElement.components to FunctionSpace.components

### PR DESCRIPTION
**Pull request by mscroggs**
Docstring is about a physical cell which the finite element is no longer aware of, but the function space is

### CODE REVIEW

File: ufl/functionspace.py [hunk 1]
```diff
@@ -60,6 +62,32 @@ def __init__(self, domain, element, label=""):
...
 
+    @property
+    def components(self) -> typing.Dict[typing.Tuple[int, ...], int]:
+        """Get the numbering of the components of the element of this space.
+
+        Returns:
+            A map from the components of the values on a physical cell (eg (0, 1))
+            to flat component numbers on the reference cell (eg 1)
+        """
+        from ufl.pullback import SymmetricPullback
```
Comment (jorgensd) [inline ufl/functionspace.py:73]: Can we move the import outside the function itself?

Comment (jorgensd) [reply ufl/functionspace.py:73]: Then I am fine with this.

**Comment by jorgensd** [comment]
@dham do you have any objections to merging this?

**Comment by dham** [comment]
I think this is a natural consequence of moving geometric quantities out of elements. I have no objection.

**Pull request merged**